### ISSUE TITLE
How to make a basic mac filter ?

### ISSUE THREAD

**Issue by TheLime1** [Salamek/huawei-lte-api#203]
For exmple i want to filter `3A:7B:56:2C:9D:E7`

**Comment by Salamek** [comment]
It should be in [https://github.com/Salamek/huawei-lte-api/blob/1a28bbf96eec2480b04e03b46584f67ffeb9dc11/huawei_lte_api/api/Security.py#L15](https://github.com/Salamek/huawei-lte-api/blob/master/huawei_lte_api/api/Security.py#L15)

But i see only adding url filter implemented right now...

**Comment by TheLime1** [comment]
i used this
https://github.com/Salamek/huawei-lte-api/blob/32226a9513cf4817cbba5be56d96576321dfc3a7/huawei_lte_api/api/WLan.py#L103-L113

this is my code 
```py
from huawei_lte_api.Client import Client
from huawei_lte_api.Connection import Connection
from huawei_lte_api.api.WLan import WLan

with Connection('http://admin:THE_Lime1@192.168.1.1/') as connection:
    client = Client(connection)
    wlan = client.wlan
    clients = [{'wifihostname': 'B612-233',
                'WifiMacFilterMac': '66:77:88:99:AA:BB'}]

    response = wlan.set_multi_macfilter_settings(clients)
```
and it give me 
```
Traceback (most recent call last):
  File "c:\Users\everp\Documents\GitHub\fixbox\getmac.py", line 11, in <module>
    response = wlan.set_multi_macfilter_settings(clients)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\everp\AppData\Local\Programs\Python\Python311\Lib\site-packages\huawei_lte_api\api\WLan.py", line 146, in set_multi_macfi
lter_settings                                                                                                                                 return self._session.post_set('wlan/multi-macfilter-settings', {
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\everp\AppData\Local\Programs\Python\Python311\Lib\site-packages\huawei_lte_api\Session.py", line 215, in post_set        
    self._post(endpoint, data, refresh_csrf, prefix, is_encrypted, is_json)
  File "C:\Users\everp\AppData\Local\Programs\Python\Python311\Lib\site-packages\huawei_lte_api\Session.py", line 34, in wrapped
    return fn(*args, **kw)
           ^^^^^^^^^^^^^^^
  File "C:\Users\everp\AppData\Local\Programs\Python\Python311\Lib\site-packages\huawei_lte_api\Session.py", line 254, in _post
    response_data = cast(str, self._check_response_status(self._process_response_data(response)))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\everp\AppData\Local\Programs\Python\Python311\Lib\site-packages\huawei_lte_api\Session.py", line 156, in _check_response_
status                                                                                                                                        raise error_code_to_exception.get(error_code, ResponseErrorException)(
huawei_lte_api.exceptions.ResponseErrorException: 9003: Unknown
```

**Comment by TheLime1** [comment]
when i click **save**
![image](https://github.com/Salamek/huawei-lte-api/assets/47940043/730df04a-0488-4215-be45-0a3e7f63249f)

**Comment by TheLime1** [comment]
and using this 
```py
from huawei_lte_api.Client import Client
from huawei_lte_api.Connection import Connection
from huawei_lte_api.api.WLan import WLan

with Connection('http://admin:THE_Lime1@192.168.1.1/') as connection:
    client = Client(connection)
    wlan = client.wlan
    wlan.set_mac_filter('myhostname', '66:77:88:99:AA:BB')
```
give
```
tHub/fixbox/getmac.py                                                                                                                     Traceback (most recent call last):
  File "c:\Users\everp\Documents\GitHub\fixbox\getmac.py", line 8, in <module>
    wlan.set_mac_filter('myhostname', '66:77:88:99:AA:BB')
  File "C:\Users\everp\AppData\Local\Programs\Python\Python311\Lib\site-packages\huawei_lte_api\api\WLan.py", line 159, in set_mac_filter 
    return self._session.post_set('wlan/mac-filter', OrderedDict((
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\everp\AppData\Local\Programs\Python\Python311\Lib\site-packages\huawei_lte_api\Session.py", line 215, in post_set        
    self._post(endpoint, data, refresh_csrf, prefix, is_encrypted, is_json)
  File "C:\Users\everp\AppData\Local\Programs\Python\Python311\Lib\site-packages\huawei_lte_api\Session.py", line 34, in wrapped
    return fn(*args, **kw)
           ^^^^^^^^^^^^^^^
  File "C:\Users\everp\AppData\Local\Programs\Python\Python311\Lib\site-packages\huawei_lte_api\Session.py", line 254, in _post
    response_data = cast(str, self._check_response_status(self._process_response_data(response)))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\everp\AppData\Local\Programs\Python\Python311\Lib\site-packages\huawei_lte_api\Session.py", line 156, in _check_response_
status                                                                                                                                        raise error_code_to_exception.get(error_code, ResponseErrorException)(
huawei_lte_api.exceptions.ResponseErrorNotSupportedException: 100002: No support
```

**Comment by TheLime1** [comment]
i see a pattern with this erorr code 
https://github.com/Salamek/huawei-lte-api/issues?q=is%3Aissue+is%3Aopen+100002

**Comment by Salamek** [comment]
You have used wrong function, your router is using `set_multi_macfilter_settings` call, try that

**Comment by TheLime1** [comment]
> You have used wrong function, your router is using set_multi_macfilter_settings call, try that

> this is my code
> 
> ```python
> from huawei_lte_api.Client import Client
> from huawei_lte_api.Connection import Connection
> from huawei_lte_api.api.WLan import WLan
> 
> with Connection('http://admin:THE_Lime1@192.168.1.1/') as connection:
>     client = Client(connection)
>     wlan = client.wlan
>     clients = [{'wifihostname': 'B612-233',
>                 'WifiMacFilterMac': '66:77:88:99:AA:BB'}]
> 
>     response = wlan.set_multi_macfilter_settings(clients)
> ```
> 
> and it give me
> 
> ```
> Traceback (most recent call last):
>   File "c:\Users\everp\Documents\GitHub\fixbox\getmac.py", line 11, in <module>
>     response = wlan.set_multi_macfilter_settings(clients)
>                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "C:\Users\everp\AppData\Local\Programs\Python\Python311\Lib\site-packages\huawei_lte_api\api\WLan.py", line 146, in set_multi_macfi
> lter_settings                                                                                                                                 return self._session.post_set('wlan/multi-macfilter-settings', {
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "C:\Users\everp\AppData\Local\Programs\Python\Python311\Lib\site-packages\huawei_lte_api\Session.py", line 215, in post_set        
>     self._post(endpoint, data, refresh_csrf, prefix, is_encrypted, is_json)
>   File "C:\Users\everp\AppData\Local\Programs\Python\Python311\Lib\site-packages\huawei_lte_api\Session.py", line 34, in wrapped
>     return fn(*args, **kw)
>            ^^^^^^^^^^^^^^^
>   File "C:\Users\everp\AppData\Local\Programs\Python\Python311\Lib\site-packages\huawei_lte_api\Session.py", line 254, in _post
>     response_data = cast(str, self._check_response_status(self._process_response_data(response)))
>                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "C:\Users\everp\AppData\Local\Programs\Python\Python311\Lib\site-packages\huawei_lte_api\Session.py", line 156, in _check_response_
> status                                                                                                                                        raise error_code_to_exception.get(error_code, ResponseErrorException)(
> huawei_lte_api.exceptions.ResponseErrorException: 9003: Unknown
> ```

**Comment by Salamek** [comment]
Is that correct payload? Check what is your router sending in browser debug tools "Payload" card...

**Comment by Salamek** [comment]
Closed by #204

**Comment by Salamek** [comment]
Released in 1.8.1
